op-node/rollup/derive/span_batch.go

```diff
@@ -544,6 +553,33 @@ func (b *SpanBatch) AppendSingularBatch(singularBatch *SingularBatch, seqNum uin
 	return b.sbtxs.AddTxs(newTxs, b.ChainID)
 }
 
+func (b *SpanBatch) revertLastBatch() {
+	if len(b.Batches) == 0 {
+		panic("revertLastBatch should only be called when there's at least 1 batch")
+	}
+	// revert ParentCheck
+	copy(b.ParentCheck[:], b.lastParentCheck[:])
```

Comment:
What happens if a caller attempts to revert the last batch multiple times? Looks like `lastParentCheck` will remain, and while the batches will roll back, this value will persist.

You'd need to keep an entire slice of `ParentCheck`s in order to support reverting more than once, right?

Reply:
I added a comment that the revert call should be paired with `AppendSingularBatch`.